src/inmanta/data/__init__.py

```diff
@@ -1567,7 +1567,7 @@ def to_dto(self) -> m.CompileRun:
             force_update=self.force_update,
             metadata=self.metadata,
             environment_variables=self.environment_variables,
-            compile_data=None if self.compile_data is None else json.loads(self.compile_data),
+            compile_data=None if self.compile_data is None else m.CompileData(**self.compile_data),
```

Comment:
Not strictly required to convert it here but I also see no reason not to.

Reply:
the typing is not correct otherwise,...